### PR TITLE
[WIP] Initial API Versioning | Github Integration

### PR DESCRIPTION
**Pull request by React1-X**

### CODE REVIEW

File: server/src/app.routes.ts [hunk 1]
```diff
@@ -40,17 +46,17 @@ router.post("/base/save", auth, debugDataLog("Base save data"), baseSave);
...
-router.post("/api/bm/base/infernomonsters", auth, debugDataLog("Load inferno monsters"), infernoMonsters);
-router.post("/api/bm/base/save", auth, debugDataLog("Inferno save data"), baseSave);
+router.post("/api/:apiVersion/bm/base/load", apiVersion, auth, debugDataLog("Inferno load data"), baseLoad);
+router.post("/api/:apiVersion/bm/base/infernomonsters", apiVersion, auth, debugDataLog("Load inferno monsters"), infernoMonsters);
+router.post("/api/:apiVersion/bm/base/save", apiVersion, auth, debugDataLog("Inferno save data"), baseSave);
 
 // Worldmap v2
 router.post("/worldmapv2/getarea", auth, debugDataLog("MR2 get area"), getArea);
-router.post("/api/player/savebookmarks", auth, debugDataLog("MR2 save bookmarks"), getArea);
+router.post("/api/:apiVersion/player/savebookmarks", auth, debugDataLog("MR2 save bookmarks"), getArea);
```
Comment (EvenBiggerManGhark) [inline server/src/app.routes.ts:59]: MIssing apiVersion middleware

File: server/src/mikro-orm.config.ts [hunk 2]
```diff
@@ -3,13 +3,14 @@ import { MariaDbDriver } from "@mikro-orm/mariadb";
...
 import { User } from "./models/user.model";
+import { ENV } from "./enums/Env";
 
 // The configuration for the ORM - Any Entities added need to be put in here, other than that probably doesn't need to be touched
 // tslint:disable-next-line: no-object-literal-type-assertion
 export default {
   type: "mariadb",
   allowGlobalContext: false,
-  debug: process.env.ENV !== "production" ? true : false,
+  debug: process.env.ENV !== ENV.PROD ? true : false,
```
Comment (EvenBiggerManGhark) [inline server/src/mikro-orm.config.ts:13]: We dont need the ternary here process.env.ENV !== ENV.PROD evaluates to true or false

File: server/src/app.routes.ts (hunk 1 above)
Comment (React1-X) [reply server/src/app.routes.ts]: Fixed in [latest commit](https://github.com/bym-refitted/backyard-monsters-refitted/commit/4d306e150f1f7dfb4c92ef94d6f2c0084345cb08)

File: server/src/mikro-orm.config.ts (hunk 2 above)
Comment (React1-X) [reply server/src/mikro-orm.config.ts]: Fixed in [latest commit](https://github.com/bym-refitted/backyard-monsters-refitted/commit/4d306e150f1f7dfb4c92ef94d6f2c0084345cb08)

**Pull request merged**